crowbar_framework/lib/runtime_checks.rb

```diff
@@ -0,0 +1,166 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "resolv"
+require "ipaddr"
+require "socket"
+require "logger"
+
+module RuntimeChecks
+  class NetworkChecks
+    class << self
+      def hostname
+        `hostname`.gsub("\n", "")
+      end
+
+      def domain
+        `hostname -d`.gsub("\n", "")
+      end
+
+      def fqdn
+        `hostname -f`.gsub("\n", "")
+      end
+
+      def ipv4_addrs(fqdn)
+        ipv4_addrs = []
+
+        Resolv.getaddresses(fqdn).each do |address|
+          ip_addr = IPAddr.new(address)
+          ipv4_addrs.push address if ip_addr.ipv4?
+        end
+
+        ipv4_addrs
+      end
+
+      def ipv6_addrs(fqdn)
+        ipv6_addrs = []
+
+        Resolv.getaddresses(fqdn).each do |address|
+          ip_addr = IPAddr.new(address)
+          ipv6_addrs.push address if ip_addr.ipv6?
+        end
+
+        ipv6_addrs
+      end
+    end
+
+    def initialize(logger = Logger.new(STDOUT))
+      @logger = logger
+
+      @hostname = NetworkChecks.hostname
```

Comment:
In general I would prefer some lazy loading, this stuff needs to be executed only if we get to a check that needs that information.